### PR TITLE
TST: a couple improvements for ci_install.sh

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

Trying out a couple simplifications that I _think_ should be possible now.

### CODE REVIEW

File: tests/ci_install.sh [hunk 1]
```diff
@@ -34,26 +33,24 @@ if [[ "${RUNNER_OS}" == "Windows" ]] && [[ ${dependencies} != "minimal" ]]; then
...
-    python -m pip install -e .[test,minimal]
+    python -m pip install -e ".[test,minimal]"
 elif [[ ${dependencies} == "full" ]]; then
     # test with all optional runtime dependencies
 
     # this is required for cartopy. see
     # https://scitools.org.uk/cartopy/docs/latest/installing.html?highlight=install#building-from-source
     python -m pip install shapely --no-binary=shapely
-    CFLAGS="$CFLAGS -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H" python -m pip install -e .[test,full]
+    CFLAGS="$CFLAGS -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H" python -m pip install -e ".[test,full]"
```
Comment (neutrinoceros) [inline tests/ci_install.sh:50]: It's probably worth trying this simplification too by now
```suggestion
    python -m pip install -e ".[test,full]"
```

File: tests/ci_install.sh [hunk 2]
```diff
@@ -34,26 +33,24 @@ if [[ "${RUNNER_OS}" == "Windows" ]] && [[ ${dependencies} != "minimal" ]]; then
     while read requirement; do conda install --yes $requirement; done < tests/windows_conda_requirements.txt
 else
     python -m pip install --upgrade pip
```
Comment (neutrinoceros) [inline tests/ci_install.sh:35]: I should also add a comment here why we upgrade pip (it's needed in most Python versions to support `pip install .[full]`, see #4270)

File: tests/ci_install.sh [hunk 3]
```diff
@@ -20,10 +20,9 @@ osx|macOS)
 esac
 
 # Disable excessive output
-mkdir -p $HOME/.config/yt
-echo "[yt]" > $HOME/.config/yt/yt.toml
-echo "suppress_stream_logging = true" >> $HOME/.config/yt/yt.toml
-cat $HOME/.config/yt/yt.toml
+echo "[yt]\nlog_level = 50" > yt.toml
```
Comment (matthewturk) [inline tests/ci_install.sh:23]: This is a change from before, yes?  Previously we turned off the stream, and now we're setting the level to 50.  This *may* be a distinction without a difference.

File: tests/ci_install.sh [hunk 4]
```diff
@@ -33,27 +32,28 @@ if [[ "${RUNNER_OS}" == "Windows" ]] && [[ ${dependencies} != "minimal" ]]; then
...
 # but the primary intention is to embed this script in CI jobs
 if [[ ${dependencies} == "minimal" ]]; then
     # test with minimal versions of runtime dependencies
-    python -m pip install -e .[test,minimal]
+    python -m pip install -e ".[test,minimal]"
 elif [[ ${dependencies} == "full" ]]; then
     # test with all optional runtime dependencies
 
     # this is required for cartopy. see
     # https://scitools.org.uk/cartopy/docs/latest/installing.html?highlight=install#building-from-source
```
Comment (matthewturk) [inline tests/ci_install.sh:51]: does this comment need updating?

**Review by matthewturk — APPROVED** [review]

File: tests/ci_install.sh (hunk 3 above)
Comment (neutrinoceros) [reply tests/ci_install.sh:23]: yes it's a intentional change of behaviour. I don't think we have any errors logged in tests but if we do, I figure we might give ourselves a chance to see them

File: tests/ci_install.sh (hunk 4 above)
Comment (neutrinoceros) [reply tests/ci_install.sh:51]: It doesn't. Actually it's the linked documentation that's out of date (see https://github.com/yt-project/yt/pull/4311#pullrequestreview-1271134788) ;-)

**Comment by neutrinoceros** [comment]
@matthewturk are you satisfied with my responses ?

**Pull request merged**